homeassistant/components/tuya/__init__.py

```diff
@@ -108,12 +108,18 @@ async def async_setup(hass, config):
     """Set up the Tuya integration."""
 
     conf = config.get(DOMAIN)
-    if conf is not None:
-        hass.async_create_task(
-            hass.config_entries.flow.async_init(
-                DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
-            )
+    if conf is None:
+        return True
+
+    domains_list = hass.config_entries.async_domains()
+    if DOMAIN in domains_list:
+        return True
```

Comment:
This should be handled in the configuration flow, not here.

Furthermore, this PR makes 2 unrelated changes, which should generally be voided.

I would advise you to revert the part in this file.

Reply:
It was suggested to me to do this check here in another PR, but I agree with you that this check already exist in configuration flow. It is ok if I leave the first check as is that is more clear?

```
    if conf is None:
        return True
```

Reply:
I think it is unrelated to the change you are making in this PR.

Reply:
Ok, change reverted.